chatbot/middleware/rate_limiting.py

```
@@ -0,0 +1,368 @@
+            django.http.HttpResponse or None: در صورت شناسایی محتوای حساس، یک JsonResponse با وضعیت 400 بازگردانده می‌شود، در غیر این صورت None تا پردازش ادامه یابد.
+        """
+        if not self._is_chatbot_api(request.path):
+            return None
+        
+        # بررسی محتوای حساس در درخواست
+        if request.method == 'POST' and hasattr(request, 'body'):
+            try:
+                body_content = request.body.decode('utf-8').lower()
+                if self._contains_sensitive_content(body_content):
+                    logger.warning(
+                        f"Sensitive content detected from user {request.user.id if request.user.is_authenticated else 'anonymous'}"
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)